_ingest-pipelines/complex-conditionals.md

```diff
@@ -0,0 +1,442 @@
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "level": "critical"
+        },
+        "_ingest": {
+          "timestamp": "2025-04-23T10:39:25.468434835Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+## Multi-conditional processing
+
+The following pipeline:
+
+- adds a field `env` set to `production` if field `env` doesn't already exist.
+- adds a field `severity` set to `major` if `status` field is higher or equal to `500`.
+- drops the document if `env` field is set to `test` and `message` field contains `debug`.
+
+```json
+PUT _ingest/pipeline/advanced_log_pipeline
+{
+  "processors": [
+    {
+      "set": {
+        "field": "env",
+        "value": "production",
+        "if": "!ctx.containsKey('env')"
+      }
+    },
+    {
+      "set": {
+        "field": "severity",
+        "value": "major",
+        "if": "ctx.status != null && ctx.status >= 500"
+      }
+    },
+    {
+      "drop": {
+        "if": "ctx.env == 'test' && ctx.message?.contains('debug')"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Use the following `_simulate` request to test the pipeline:
+
+```json
+POST _ingest/pipeline/advanced_log_pipeline/_simulate
+{
+  "docs": [
+    {
+      "_source": {
+        "status": 503,
+        "message": "Server unavailable"
+      }
+    },
+    {
+      "_source": {
+        "env": "test",
+        "message": "debug log output"
+      }
+    },
+    {
+      "_source": {
+        "status": 200,
+        "message": "OK"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Expected output:
+- First document has fields `env: production` and `severity: major` added.
+- Second document is dropped.
+- Third document has field `env: production` added.
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "severity": "major",
+          "message": "Server unavailable",
+          "env": "production",
+          "status": 503
+        },
+        "_ingest": {
+          "timestamp": "2025-04-23T10:51:46.795026554Z"
+        }
+      }
+    },
+    null,
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "message": "OK",
+          "env": "production",
+          "status": 200
+        },
+        "_ingest": {
+          "timestamp": "2025-04-23T10:51:46.795048304Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+## Null-safe notation
+
+Null-safe navigation notation (`?.`) should be used to check if the field is `null`, however, this notation can return `null` silently, therefore it is recommended to first check if the returned value is `null` and then use operations like `.contains` or `==`.
+
+Not safe syntax:
+
+```
+"if": "ctx.message?.contains('debug')"
+```
+
+If the field `message` does not exist in the document, this will return `null_pointer_exception` with error `Cannot invoke "Object.getClass()" because "value" is null`
```

Comment:
```suggestion
If the `message` field does not exist in the document, this request returns a `null_pointer_exception` with a message `Cannot invoke "Object.getClass()" because "value" is null`.
```